microros_olimex_e407_extensions/Src/allocators.c

```diff
@@ -14,17 +14,23 @@ void * __freertos_allocate(size_t size, void * state){
 void __freertos_deallocate(void * pointer, void * state){
   (void) state;
   // printf("-- Free %d (prev: %d B)\n",getBlockSize(pointer), xPortGetFreeHeapSize());
-  usedMemory -= getBlockSize(pointer);
-  vPortFree(pointer);
+  if (NULL != pointer){
+    usedMemory -= getBlockSize(pointer);
+    vPortFree(pointer);
+  }
 }
 
 void * __freertos_reallocate(void * pointer, size_t size, void * state){
   (void) state;
   // printf("-- Realloc %d -> %d (prev: %d B)\n",getBlockSize(pointer),size, xPortGetFreeHeapSize());
   absoluteUsedMemory += size;
   usedMemory += size;
-  usedMemory -= getBlockSize(pointer);
-  return pvPortRealloc(pointer,size);
+  if (NULL != pointer){
```

Comment:
For readability, I propose to swap the two cases, in order to avoid an else-branch for a unequal comparison.